aws/structure.go

```diff
@@ -1088,6 +1088,51 @@ func flattenESClusterConfig(c *elasticsearch.ElasticsearchClusterConfig) []map[s
 	return []map[string]interface{}{m}
 }
 
+func expandESCognitoOptions(m map[string]interface{}) *elasticsearch.CognitoOptions {
+	options := elasticsearch.CognitoOptions{}
+
+	if cognitoEnabled, ok := m["enabled"]; ok {
+		options.Enabled = aws.Bool(cognitoEnabled.(bool))
+
+		if cognitoEnabled.(bool) {
+
+			if v, ok := m["user_pool_id"]; ok && v.(string) != "" {
+				options.UserPoolId = aws.String(v.(string))
+			}
+			if v, ok := m["identity_pool_id"]; ok && v.(string) != "" {
+				options.IdentityPoolId = aws.String(v.(string))
+			}
+			if v, ok := m["role_arn"]; ok && v.(string) != "" {
+				options.RoleArn = aws.String(v.(string))
+			}
+		}
+	}
+
+	return &options
+}
+
+func flattenESCognitoOptions(c *elasticsearch.CognitoOptions) []map[string]interface{} {
+	m := map[string]interface{}{}
+
+	if c.Enabled != nil {
+		m["enabled"] = *c.Enabled
+	}
+
+	if aws.BoolValue(c.Enabled) {
+		if c.UserPoolId != nil {
```

Comment:
Nitpick: We can remove the `nil` checks and still prevent potential panics at the same time by using the SDK provided functions like `aws.StringValue()` -- since the schema will default to `""` for `schema.TypeString` and `aws.StringValue()` returns `""` for `nil`.

```go
m["identity_pool_id"] = aws.StringValue(c.IdentityPoolId)
m["user_pool_id"] = aws.StringValue(c.UserPoolId)
m["role_arn"] = aws.StringValue(c.RoleArn)
```